### PR TITLE
Add `otelcol.connector.host_info` component

### PR DESCRIPTION
**Pull request by rlankfo**
#### PR Description
Introduce `otelcol.connector.host_info` component.

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [ ] Config converters updated

### CODE REVIEW

File: docs/sources/flow/reference/components/otelcol.connector.grafanacloud.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  stage: experimental
+title: otelcol.connector.grafanacloud
+---
+
+# otelcol.connector.grafanacloud
+
+{{< docs/shared lookup="flow/stability/experimental.md" source="agent" version="<AGENT_VERSION>" >}}
+
+`otel.connector.grafanacloud` accepts span data from other `otelcol` components and generates usage
+metrics.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.connector.grafanacloud.md:17]:
```suggestion
`otel.connector.grafanacloud` accepts span data from other `otelcol` components and generates usage metrics.
```

Don't need the hard return at 80 chars... bring sentence back together in one line

**Review by clayton-cornell — COMMENTED** [review]
Minor doc tweak

File: component/otelcol/connector/grafanacloud/factory.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package grafanacloud
```
Comment (rfratto) [inline component/otelcol/connector/grafanacloud/factory.go:1]: IMO it's a bit awkward for a native Flow component to go for the otelcol factory + wrapper approach, which I'd expect to only see if we were wrapping a component made for the pure collector engine. 

If possible, can this be reduced to just a Flow component? That would help cut down the amount of code. For examples,  [otelcol.receiver.prometheus](https://github.com/grafana/agent/blob/main/component/otelcol/receiver/prometheus/prometheus.go) and [otelcol.exporter.prometheus](https://github.com/grafana/agent/blob/main/component/otelcol/exporter/prometheus/prometheus.go) are two native Flow components that deal with OpenTelemetry data.

Comment (rlankfo) [reply component/otelcol/connector/grafanacloud/factory.go:1]: The plan  is to contribute this component upstream to opentelemetry-collector-contrib so otel collector users can use this connector. We didn't want to block the rollout of this component on having it upstreaming first. The plan is to get this in for the next release, port it to contrib, and then import it from contrib at which time we'd be able to remove the connector implementation.

File: component/otelcol/connector/grafanacloud/connector.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/connector"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+const (
+	hostInfoMetric     = "host_info"
```
Comment (jcreixell) [inline component/otelcol/connector/grafanacloud/connector.go:16]: shall we call this `trace_host_info`? rationale: we may want to keep separate metrics for each of the telemetry signals (so far we only look at traces to extract this information).

File: component/otelcol/connector/grafanacloud/connector.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+}
+
+// Capabilities implements connector.Traces.
+func (c *connectorImp) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+// ConsumeTraces implements connector.Traces.
+func (c *connectorImp) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
+	// loop through the levels of spans of the one trace consumed
```
Comment (jcreixell) [inline component/otelcol/connector/grafanacloud/connector.go:52]: this comment is a leftover from the example connector in https://opentelemetry.io/docs/collector/building/connector/, we can keep it or remove it

File: component/otelcol/connector/grafanacloud/connector.go [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+	return nil
+}
+
+func (c *connectorImp) flush(ctx context.Context) error {
+	var err error
+
+	metrics, count := c.hostMetrics.metrics()
+	if count > 0 {
+		c.hostMetrics.reset()
+		c.logger.Info("Flushing metrics", zap.Int("count", count))
```
Comment (jcreixell) [inline component/otelcol/connector/grafanacloud/connector.go:116]: shall we make this debug level?

**Review by jcreixell — APPROVED** [review]
added some comments, overall LGTM

File: component/otelcol/connector/grafanacloud/factory.go (hunk 2 above)
Comment (rfratto) [reply component/otelcol/connector/grafanacloud/factory.go:1]: OK 👍

File: component/otelcol/connector/grafanacloud/grafanacloud.go [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+	"github.com/grafana/agent/component/otelcol"
+	"github.com/grafana/agent/component/otelcol/connector"
+	"github.com/grafana/river"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	otelextension "go.opentelemetry.io/collector/extension"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "otelcol.connector.grafanacloud",
```
Comment (rfratto) [inline component/otelcol/connector/grafanacloud/grafanacloud.go:18]: FWIW: our naming convention for components expects multiple words to be separated by underscores; for this to align with the convention this should be `otelcol.connector.grafana_cloud`

Comment (rlankfo) [reply component/otelcol/connector/grafanacloud/grafanacloud.go:18]: thanks @rfratto. It looks like with otelcol components, there may be some inconsistency. If we wrap contrib components then do we want to match the names? contrib omits the underscores.

Comment (rfratto) [reply component/otelcol/connector/grafanacloud/grafanacloud.go:18]: We've noticed the inconsistency for sure :) It's something we're looking into resolving. Matching the upstream components exactly isn't necessary; Flow components are already named differently (with the namespaces) and we can use documentation to inform people about what the wrapped component is called.

Comment (rlankfo) [reply component/otelcol/connector/grafanacloud/grafanacloud.go:18]: Sounds good! once the final name is decided, I'll make sure we follow the conventions here.

**Review by clayton-cornell — APPROVED** [review]

File: docs/sources/flow/reference/components/otelcol.connector.host_info.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+{{< docs/shared lookup="flow/stability/experimental.md" source="agent" version="<AGENT_VERSION>" >}}
+
+`otel.connector.host_info` accepts span data from other `otelcol` components and generates usage metrics.
+
+## Usage
+
+```river
+otelcol.connector.host_info "LABEL" {
+  output {
```
Comment (thampiotr) [inline docs/sources/flow/reference/components/otelcol.connector.host_info.md:22]: This output block is not documented for some reason? cc @jcreixell

Comment (rlankfo) [reply docs/sources/flow/reference/components/otelcol.connector.host_info.md:22]: thanks for catching this @thampiotr. opened a pr to document this block: https://github.com/grafana/agent/pull/6488

**Pull request merged**